### ISSUE TITLE
Convert queued videos in a playlist

### ISSUE THREAD

**Issue by tampos4** [libre-tube/LibreTube#2665]
### Describe your suggested feature

One of the feature missing in LT which I used to use regularly in NewPipe is to be able to convert queued videos into a playlist.

Sometimes we just queue video after video and then we don't have mood or time or both to watch those videos.

Then an option (beside shuffle or repeat icon)  to convert those queued videos into a playlist can be very helpful.
That option can add those videos into an existing playlist or create a new playlist.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
Looks like a good idea to me!

**Comment by Bnyro** [comment]
Any suggestions where to place a button for it? Eventually we would also need to reorder the buttons at the bottom of the playing queue since there are currently two per side, adding another one to one side would make it look asymmetric.

**Comment by tampos4** [comment]
> ![Screenshot_20230114-001411_LibreTube Debug](https://user-images.githubusercontent.com/121803062/212411120-6994d24f-8d7c-49ad-86a7-b39d5032596b.jpg)

What is the use of this button (second button from the left) ?

I mean I know what it does.

If the currently playing video is at the top then it moves that video to the bottom of the queue.
If the currently playing video is at 2nd position from top then it sends the video to the 2nd position from bottom.
If the currently playing video is at 4th from the bottom then it send the video to 4th from the top.

~~Why does anyone use that ?~~
Does any use that?

• If yes, then -

1. the "Add to Playlist" button can be placed beside the mid button
OR
2. the "Add to Playlist" button can replace the mid button as most of us (i guess) just slide down the queue to close it rather than first sliding up to go to the bottom of the queue then clicking on mid button to close the queue.

• If no, then that button (second from the left) can be replaced with "Add to Playlist" button.

Btw here is NewPipe's placement.

![IMG_20230114_014036](https://user-images.githubusercontent.com/103326242/212413995-8a12998e-eccd-4821-8436-23567cd3d5d9.jpg)

**Comment by tampos4** [comment]
Also I just noticed, the cross button (×) in LT removes all the videos from the queue which are at the bottom of the currently playing video.
Why?

If I don't want any video to play after the currently playing video then I will simply close the video after watching it.
OR
If I want to add some other videos to the queue in place of those videos then I will simply add those videos.

[In my personal opinion] removing those videos is redundant. And also it causes an extra button.

The cross button (×) in NewPipe closes the queue which in LT is the mid button's job.

tl;dr
> we would also need to reorder the buttons at the bottom of the playing queue

YES

I think the cross button (×) button in LT should do what its original job is as it does in NewPipe i.e., to close the queue.
And the new "Add to Playlist" button can sit alone at the middle.

**Comment by Bnyro** [comment]
The button to reverse the queue is used by anyone, yes, it was a feature request so at least one person does use it :)
The best solution is probably to get rid of the "X" button as you said and replace it with the playlist button.

**Comment by tampos4** [comment]
Thanks. Works flawlessly.

But..
This (i guess) introduced a new bug.

Now the queued videos remain unchanged while playing different videos on different topics.

However a restart fixes that.

Should I create a new issue for this?

**Comment by Bnyro** [comment]
I don't really get the issue but it's not related to this issue here, so better move to an other issue.
